### ISSUE TITLE
add an option for system include paths

### ISSUE THREAD

**Issue by mycoboco** [mycoboco/beluga#33]
Headers from the system include paths need to silence warnings from `sc` and `beluga`.

**Comment by mycoboco** [comment]
https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html

**Comment by mycoboco** [comment]
- decided not to implement `-sysroot`, `-isysroot` and replacement of `=`.
- decided not to implement `-iquote` because it looks inconsistent with the intent of the standard.
- decided to implement:
  - separation of the `-I` path list from the system include path list;
  - removal of duplicated paths
  - `-nostdinc`
- the order of include paths:
  - from `-I`
  - from `-isystem`
  - from environment variables
  - from `-idirafter`

**Comment by mycoboco** [comment]
5be2cebb551925d53973cd3adf1e5233ede465e4
(to add a reference to this issue in that commit)

**Comment by mycoboco** [comment]
"prohibition of adding a system include path as normal one" is subsumed by "removal of duplicated paths".

**Comment by mycoboco** [comment]
Some observations:

Removal of duplicated paths will work properly with `HAVE_REALPATH` defined; simple string match is performed without it and relative paths that refer to the same location cannot be detected.

**Comment by mycoboco** [comment]
All features described above have been implemented by e7201f61a3dcc81a197338ce99dafd5deef6d7e5. An internal option to silence warnings remained.

**Comment by mycoboco** [comment]
Failed to handle warnings that are triggered by system headers after having seen the headers, e.g.,

```C
/* foo.h; system header */
static int unused;

/* foo.c */
#include <foo.h>
```
